pymc3/distributions/discrete.py

```diff
@@ -285,8 +285,13 @@ def __init__(self, p, *args, **kwargs):
     def logp(self, value):
         p = self.p
         k = self.k
-
-        return bound(log(p[value]),
+        
+        try:
+            pv = log(p[value])
+        except IndexError:
+            pv = log(constant(p)[value])
```

Comment:
If the `value` is a variable, it cannot be used to index a Numpy array, so it generates an index error. This is only for the case where `p` is a numpy array or list and `value` is a Tensor.